### DISCUSSION TITLE
"jump" to the "last" commit

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#4538] [Q&A] [answered]
When switching between numerous commits (or branches) with different work on them, is there an easy way to navigate "backward" to find the commit you were on before the last `new`, or before a specific operation in the `op log`?

For instance, say I have two branches, `b1` and `b2`. `b1` is a bit out of date and then I rebase it, but it has conflicts starting a couple commits into the branch, so I use `jj new <first-conflict>`. I resolve some, but not all, of the conflicts. I then, for whatever reason, jump to `b2`, maybe to quickly resolve a simple CI issue (hahahaha).

I then want to jump back to `<first-conflict>-`, but I don't necessarily remember what this commit was called. My shell history is helpful but not sufficient; for instance, maybe I only needed the first letter of `<first-conflict>` to check it out the first time, but now that my working copy is at `b2`, that's no longer a valid ref.

I realize I could do something like `jj op log`, find the operation, then `jj log --at-operation <op> -r @`, but is there a nicer way?

Note: `jj prev` is based on the commit graph, not the operation log, so that's not what I'm looking for.

**Comment by yuja** [comment, marked as answer]
https://github.com/martinvonz/jj/issues/2871 is probably what you want. It's not implemented yet.
